sdk/appconfiguration/app-configuration/review/app-configuration.api.md

```diff
@@ -74,60 +76,24 @@ export interface DeleteConfigurationSettingOptions extends HttpOnlyIfUnchangedFi
 export interface DeleteConfigurationSettingResponse extends SyncTokenHeaderField, HttpResponseFields, HttpResponseField<SyncTokenHeaderField> {
 }
 
-// @public
-export interface FeatureFlag extends FeatureFlagParam, ConfigurationSetting {
-}
-
 // @public
 export const featureFlagContentType = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8";
 
 // @public
-export interface FeatureFlagParam extends ConfigurationSettingParam {
+export const featureFlagPrefix = ".appconfig.featureflag/";
+
+// @public (undocumented)
+export interface FeatureFlagValue {
     conditions: {
-        clientFilters: (FeatureFlagTargetingClientFilter | FeatureFlagTimeWindowClientFilter | FeatureFlagPercentageClientFilter | Record<string, unknown>)[];
+        clientFilters: {
+            name: string;
+            parameters?: Record<string, unknown>;
```

Comment:
@xirzec @pakrym

Reply:
Hmm I wonder... we aren't strong-typing, but could we write a small function that `toLowerCase`s the property names?

Reply:
Does the portal preserve whatever casing is passed in? I assumed the service always used PascalCase, so we'd be converting PascalCase into camelCase and back when sending it to the server (much like we do with mappers.)

Reply:
> Does the portal preserve whatever casing is passed in?

Haven't tested, but I think so, yes!

> so we'd be converting PascalCase into camelCase and back when sending it to the server

```js
// Expected by the service
{
    "id": "name-1161727070078303372-2",
    "description": "I'm new description",
    "enabled": false,
    "conditions": {
        "client_filters": [
            {
                "name": "Microsoft.TimeWindow",
                "parameters": {
                    "Start": "Wed, 01 May 2019 13:59:59 GMT",
                    "End": "Mon, 01 July 2019 00:00:00 GMT"
                }
            },
            {
                "name": "FilterX"
            }
        ]
    }
}

// (Property names following the TS guidelines)
// Input from a JS user
// Our strongly typed objects would suggest the users to provide an object such as below
{
    "id": "name-1161727070078303372-2",
    "description": "I'm new description",
    "enabled": false,
    "conditions": {
        "clientFilters": [
            {
                "name": "Microsoft.TimeWindow",
                "parameters": {
                    "start": "Wed, 01 May 2019 13:59:59 GMT",
                    "end": "Mon, 01 July 2019 00:00:00 GMT"
                }
            },
            {
                "name": "FilterX"
            }
        ]
    }
}
```

The current approach is that we'd have strong types until the "name" and "parameters" level.
Inside the "parameters", the props must be as expected by the service because we are not providing strong types(all languages).
In the future, if we ever add strong types for the client filters, for example, we'll live with "Start" as the parameter instead of "start" to avoid breaking changes.

Reply:
Resolving as per the offline discussion